### DISCUSSION TITLE
AdminLTE vs Pure Bootstrap for a new web project?

### DISCUSSION THREAD

**Discussion by prahladyeri** [twbs/bootstrap#37176] [Q&A] [answered]
I'm primarily a freelance backend dev and for the first time venturing on full-stack development of a non-trivial web app on my own, hence I needed some guidance.

I've got all the backend stuff in php/mysql covered, I just want to know what's the best way to create a dashboard (with left sidebar) considering various aspects like long-term code maintenance and support, robustness, etc. Looks don't matter that much as it's a CRUD app but obviously, better is more appreciated.

Based on research until now, [AdminLTE](https://adminlte.io/) seems to be the most popular way of doing it among most devs although a few others like material and coreui also seem to have some street cred.

But another approach I'm considering apart from AdminLTE is to just use pure bootstrap and fiddle up my own sidebar [using something like this](https://bootsnipp.com/snippets/Q0dAX). That way, I won't be tied to just one Bootstrap version which is used by AdminLTE (v4.6) and troubleshooting will be much easier through google search and StackOverflow. What do you guys think is the right approach?

**Comment by GeoSot** [comment, marked as answer]
Knowing that AdminLTE provides a nice tested solution with plenty of component choices, and 'plain' BS provides a solid base and some standard components, it comes the moment that you, and only you, can take this decision, consider it in combination of your UI needs and the tools you use.

I was on the same thoughts some years ago, but in my case I was using a php template engine that was supporting php-components usage (easy and in one place, maintained blocks & markup), and the frontend wasn't too demanding.  At that time, using webpack lazy loading for my assets, avoiding keeping a js-vendor folder full of scripts, and use only one or two script tags in my code, seemed probably a better solution for me.  

3 years later, I haven't regretted my decision, as I kept the dependencies to the minimum, but yes, there have been times during js refactoring and UI addons, where I needed to implement some minor stuff myself
